### PR TITLE
Bump to C++17

### PR DESCRIPTION
**Pull request by pgrete**
## PR Summary

This PR just changes the version requirements so that we can go forward with this, but it does *not* cleanup code (there are couple todos that we previously identify to benefit from C++17 features).

## PR Checklist

- [x] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -250,13 +249,7 @@ if (ENABLE_COMPILER_WARNINGS)
 endif()
 
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-if (PARTHENON_ENABLE_CPP17)
-  set(CMAKE_CXX_STANDARD 17)
-# Ensure correct flag is used as deprecated c++1z is not supported by nvcc.
-  set(CMAKE_CXX17_STANDARD_COMPILE_OPTION "-std=c++17")
```
Comment (lroberts36) [inline CMakeLists.txt:256]: Is this no longer required?

**Review by lroberts36 — APPROVED** [review]
LGTM

File: CMakeLists.txt (hunk 1 above)
Comment (pgrete) [reply CMakeLists.txt:256]: IIRC the correct flags are now provided by Kokkos

**Review by jdolence — APPROVED** [review]

**Review by Yurlungur — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (pgrete) [reply CMakeLists.txt:256]: Just confirmed that I can compile with cuda on my machine without that line.

**Comment by pgrete** [comment]
Has anyone tried to triggered CI on Darwin (just to check if it really breaks or if it magically works)?

**Comment by Yurlungur** [comment]
Yes---it breaks. I have it on my TODO list to make something work.

**Comment by pgrete** [comment]
With two PRs depending on this one, I'll go ahead and merge now.

**Pull request merged**